mcs/tools/mono-helix-client/HelixBase.cs

```diff
@@ -0,0 +1,155 @@
+//
+// HelixBase.cs
+//
+// Authors:
+//	Alexander Köplinger  <alkpli@microsoft.com>
+//
+// Copyright (C) 2018 Microsoft
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+//
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net;
+using System.Threading.Tasks;
+using Microsoft.DotNet.Helix.Client;
+
+public class HelixBase
+{
+    protected IHelixApi _api;
+
+    public HelixBase ()
+    {
+        _api = ApiFactory.GetAuthenticated (GetEnvironmentVariable ("MONO_HELIX_API_KEY"));
+    }
+
+    protected string GetEnvironmentVariable (string variable)
+    {
+        return Environment.GetEnvironmentVariable (variable) ?? throw new ArgumentException ($"No value for '{variable}'.");
+    }
+
+    protected string McUrlEncode (string input)
+    {
+        // encodes the URL in a way Mission Control understands (% replaced with ~)
+        var result = WebUtility.UrlEncode (input);
+        return result.Replace ("%", "~");
+    }
+
+    public async Task<bool> WaitForJobCompletion (string correlationId)
+    {
+        bool success = false;
+        bool printedMcLink = false;
+
+        Console.WriteLine ($"Waiting for job '{correlationId}' to finish...");
+
+        int sleepTime = 0;
+
+        while (true)
+        {
+            await Task.Delay (sleepTime);
```

Comment:
```suggestion
            await Task.Delay (sleepTime).ConfigureAwait (false);
```

Reply:
This is only used in a console app so there's no synchronisation context and so the ConfigureAwait isn't needed. :)